Common/SecurityIdentifier.cs

```diff
@@ -404,6 +404,40 @@ public static SecurityIdentifier GenerateEquity(DateTime date, string symbol, st
             return Generate(date, symbol, SecurityType.Equity, market);
         }
 
+        /// <summary>
+        /// Generates the <see cref="Symbol"/> property for <see cref="QuantConnect.SecurityType.Base"/> security identifiers
+        /// </summary>
+        /// <param name="dataType">The base data custom data type if namespacing is required, null otherwise</param>
+        /// <param name="symbol">The ticker symbol</param>
+        /// <returns>The value used for the security identifier's <see cref="Symbol"/></returns>
+        public static string GenerateBaseSymbol(Type dataType, string symbol)
```

Comment:
Wondering if it's worth to make this method `public` - Seems to be used just by `GenerateBase()` and testing, maybe we could test its behavior asserting on the result of `GenerateBase()`

Reply:
I think this makes sense. This will force consumer to use the overload that requires specifying the `dataType` so the consumer is at least _forced_ to make a decision there (whether to pass `null` or some relevant `Type` instance). 👍  Great comments 👍